plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -217,20 +218,33 @@ protected String renderFeedResults(final FeedResult result)
     }
 
     if (result.getSkipLinkEntry() != null) {
-      xml.append("  <link rel=\"next\" href=\"").append(result.getBase()).append('/').append(result.getOperation());
-      xml.append("()?").append(ODataFeedUtils.skipLink(result.getSkipLinkEntry(), result.getQuery())).append("\"/>\n");
+      final String skipLink = odataLink(result.getBase(),
+          result.getOperation(),
+          ODataFeedUtils.skipLinkQueryString(result.getQuery()));
+
+      xml.append("  <link rel=\"next\" href=\"")
+          .append(XmlEscapers.xmlAttributeEscaper().escape(skipLink))
+          .append("\"/>\n");
     }
     return xml.append("</feed>").toString();
   }
 
+  public String odataLink(final String base, final String operation, final String queryString) {
+    return String.format("%s/%s()?%s", base, operation, queryString);
+  }
+
   @Override
   public void putMetadata(final Map<String, String> metadata) {
     try (StorageTx tx = openStorageTx()) {
       final Component component = createOrUpdatePackage(tx, metadata);
-      maintainAggregateInfo(tx, metadata.get(ID));
       tx.commit();
       getRepository().facet(SearchFacet.class).put(component);
     }
+    // Separate tx is necessary for the meantime, since the re-querying orient doesn't pick up uncommitted state
```

Comment:
Because the maintenance involves modifying assets, doing it in a hook leads to all the hooks being fired again. I'm pondering alternatives, but none is grabbing me as good enough.  (Expose the hooks, hook removes itself from the list; thread-local used for a hook to detect nesting; special no-hook mode for with-hook use of storageTx; 'pre-commit' hook won't work since re-querying uncommitted state doesn't work)